Add Additional Resource Attributes for OpenTelemetry

briankahoot opened the pull request:
Fixes #7322

Adds the "deployment.environment" resource attribute to those that can be configured for OpenTelemetry. This was done as some collectors, including Datadog, require this value to properly classify traces.

Changes are simply to accept the new value and update the associated documentation and tests.

ashutosh-narkar reviewed (commented):
The change lgtm. Per [this](https://docs.datadoghq.com/opentelemetry/schema_semantics/semantic_mapping/?tab=datadogexporter#unified-service-tagging)
```
deployment.environment has been deprecated in favor of deployment.environment.name in [OpenTelemetry semantic conventions v1.27.0](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.27.0)
```
Should we use the new convention?

briankahoot commented:
I did notice that, the systems I'm using still use the old version. I suppose this is the drawback of doing it property by property, when the schema changes the code has to change as well.

Would you prefer it to be the new value?

ashutosh-narkar commented:
Should be fine for now. Currently we're pre-v1.27.0. If you add a note about this in the code or commit message body it would be helpful. Then squash your commits and we can get this in. Thanks for the contribution.

briankahoot commented:
@ashutosh-narkar sorry for the delay in responding, got distracted with other things. I have added some comments and squashed it down to one commit. Let me know if this is acceptable.

ashutosh-narkar reviewed (approved):
Thanks for the contribution!
